astropy/coordinates/builtin_frames/cirs_observed_transforms.py

```diff
@@ -32,9 +32,18 @@ def cirs_to_altaz(cirs_coo, altaz_frame):
     # we use the same obstime everywhere now that we know they're the same
     obstime = cirs_coo.obstime
 
-    usrepr = cirs_coo.represent_as(UnitSphericalRepresentation)
-    cirs_ra = usrepr.lon.to(u.radian).value
-    cirs_dec = usrepr.lat.to(u.radian).value
+    if cirs_coo.distance.unit is u.one:
+        usrepr = cirs_coo.represent_as(UnitSphericalRepresentation)
+        cirs_ra = usrepr.lon.to(u.radian).value
+        cirs_dec = usrepr.lat.to(u.radian).value
+    else:
+        # compute an "astrometric" ra/dec -i.e., the direction of the
+        # displacement vector from the observer to the target in CIRS
+        loccirs = altaz_frame.location.get_itrs(cirs_coo.obstime).transform_to(cirs_coo)
+        diffrepr = (cirs_coo.cartesian-loccirs.cartesian).represent_as(SphericalRepresentation)
```

Comment:
Performance wise I think it's identical, but you're right conceptually it makes more sense (and uses slightly less memory).  Will fix.